3-enrich/stream-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/kinesis/sources/NsqSource.scala

```diff
@@ -0,0 +1,96 @@
+/*
+* Copyright (c) 2013-2017 Snowplow Analytics Ltd.
+* All rights reserved.
+*
+* This program is licensed to you under the Apache License Version 2.0,
+* and you may not use this file except in compliance with the Apache
+* License Version 2.0.
+* You may obtain a copy of the Apache License Version 2.0 at
+* http://www.apache.org/licenses/LICENSE-2.0.
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the Apache License Version 2.0 is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+* either express or implied.
+*
+* See the Apache License Version 2.0 for the specific language
+* governing permissions and limitations there under.
+*/
+
+package com.snowplowanalytics
+package snowplow.enrich
+package kinesis
+package sources
+
+
+// Apache Commons
+import org.apache.commons.codec.binary.Base64
+
+// NSQ
+import com.github.brainlag.nsq.NSQConsumer
+import com.github.brainlag.nsq.lookup.DefaultNSQLookup
+import com.github.brainlag.nsq.NSQMessage
+import com.github.brainlag.nsq.NSQConfig
+import com.github.brainlag.nsq.callbacks.NSQMessageCallback
+import com.github.brainlag.nsq.callbacks.NSQErrorCallback
+import com.github.brainlag.nsq.exceptions.NSQException
+
+// Iglu
+import iglu.client.Resolver
+
+// Snowplow
+import common.enrichments.EnrichmentRegistry
+
+// Tracker
+import com.snowplowanalytics.snowplow.scalatracker.Tracker
+
+// Logging
+import org.slf4j.LoggerFactory
+
+/**
+  * Source to read raw events from NSQ.
+  * 
+  * @param config Configuration for NSQ
+  * @param igluResolver Instance of resolver for iglu
+  * @param enrichmentRegistry EnrichmentRegistry instance
+  * @param tracker Tracker instance 
+  */
+class NsqSource(config: KinesisEnrichConfig, 
+                igluResolver: Resolver, 
+                enrichmentRegistry: EnrichmentRegistry, 
+                tracker: Option[Tracker]
+) extends AbstractSource(config, igluResolver, enrichmentRegistry, tracker) {
+
+  lazy val log = LoggerFactory.getLogger(getClass())
+
+  /**
+    * Consumer will be started to wait new message.
+    */
+  override def run(): Unit = {
+
+    val nsqCallback = new NSQMessageCallback {
+      override def message(msg: NSQMessage) = {
+        val bytes = msg.getMessage()
+        enrichAndStoreEvents(List(bytes))
+        msg.finished()
```

Comment:
`enrichAndStoreEvents` gives back a boolean saying whether or not you should checkpoint which, in the case of NSQ, means acking the message.